### DISCUSSION TITLE
Cache with cdn doesn't work?

### DISCUSSION THREAD

**Discussion by thanhtoan1196** [vercel/next.js#20326] [Help]
- I have some static page (automatically rendered as static HTML (uses no initial props)): https://test11zz.com/dang-ky

- According to the information [here](https://nextjs.org/docs/basic-features/pages#two-forms-of-pre-rendering), I thought it would be cached in cloudflare

![image](https://user-images.githubusercontent.com/16433547/102682809-df5f5e80-41fe-11eb-8a5b-bb7e18fdcc25.png)

- But in fact, it is not cached

![image](https://user-images.githubusercontent.com/16433547/102682837-159cde00-41ff-11eb-9ccc-c14b7bf75b4c.png)

**Comment by FSaldanha** [comment]
As far as I know, there is currently no simple way to deploy a fully functional Next.js app to Cloudflare.

You would need to export [your app to static HTML](https://nextjs.org/docs/advanced-features/static-html-export) (beware of some features being disabled).

**Comment by stigkj** [comment]
Cloudflare can absolutely function as a cache, but you might need to set up caching of HTML files specifically. I think that is in "Page Rules" or something. The thing @FSaldanha talks about is a specific feature of Cloudflare, called Cloudflare Pages, and that indeed only support exported pages, i.e. you run `next build && next export` and upload the result to Cloudflare Pages.

But with Cloudflare's CDN you just let Cloudflare control your domain and then when the user accesses your domain the request will go through Cloudflare as a reverse proxy and go on to your server. Then Cloudflare will cache for as long as the cache headers on the returned content specifies.
